apps/wistia-videos/src/components/Field.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {useState, useEffect} from 'react';
+import { 
+  Grid,
+  GridItem,
+  Flex,
+  Paragraph, 
+  Pill, 
+  TextInput,
+  Spinner,
+  Card
+} from '@contentful/forma-36-react-components';
+import styled from "styled-components"
+// import { arrayMove, SortableContainer, SortableElement } from 'react-sortable-hoc';
+import { FieldExtensionSDK } from '@contentful/app-sdk';
+import { WistiaItem } from "./ConfigScreen";
+import { fetchVideos } from "../functions/getVideos";
+
+interface FieldProps {
+  sdk: FieldExtensionSDK;
+}
+
+const StyledImageContainer = styled.div`
+  img {
+    width: 100%;
+    height: 100px;
+    object-fit: cover;
+    object-position: top;
+    border-radius: 6px;
+  }
+`
+
+const Field = (props: FieldProps) => {
+  const { sdk } = props;
+  const [data, updateData] = useState<WistiaItem[] | []>([])
+  const [selectedIds, setIds] = useState<number[]>([])
+  const [dropdownData, filterDropdownData] = useState<WistiaItem[] | []>([])
+  const [loading, updateLoadingStatus] = useState(true)
+  const [error, setError] = useState(null || '')
+  sdk.window.startAutoResizer()
+
+  // Set inital state based on field values
+  useEffect(() => {
+    const fieldValues = sdk.field.getValue()
+    setIds(
+      fieldValues !== undefined && fieldValues.items.length > 0 ? 
+        fieldValues.items.map((item:WistiaItem) => item.id) :[]
+    );
+    
+    const parameters:any = sdk.parameters.installation;                            
+    (async () => {
+      const videosRequest = await fetchVideos(parameters.excludedProjects, parameters.apiBearerToken
+      ) || [];
+      if (videosRequest.response.success) {
+        updateData(videosRequest.videos || [])
+        filterDropdownData(videosRequest.videos || [])
+        updateLoadingStatus(false)
+      }
+      
+      if (videosRequest.response.success === false && videosRequest.response.error) {
+        setError(videosRequest.response.error)
+        updateLoadingStatus(false)
+      }
+    })();
+      
+  }, [sdk.field, sdk.parameters.installation])
+
+  // Function to update selected video ids
+  const updateVideoIds = (id:any) => {
```

Comment:
Could be possible to add to this function the parameters from the creation of the content model? Right now is possible to add unlimited videos even when we set in the content model a limit or just one.
 The same with the _required_, it is an empty array and is a required field that shouldn't be possible to publish the entry.